openapi-diff/src/modeler/AutoRest.Swagger/Model/Operation.cs

```diff
@@ -179,51 +176,30 @@ Operation previous
         /// <summary>
         /// Check that no parameters were removed or reordered, and compare them if it's not the case
         /// </summary>
-        /// <param name="context">Comaprision Context</param>
+        /// <param name="context">Comparison Context</param>
         /// <param name="priorOperation">Operation object of old swagger</param>
         private void CheckParameters(ComparisonContext<ServiceDefinition> context, Operation priorOperation)
         {
-            // Check that no parameters were removed or reordered, and compare them if it's not the case.
-
-            var currentRoot = context.CurrentRoot;
-            var previousRoot = context.PreviousRoot;
+            ServiceDefinition currentRoot = context.CurrentRoot;
+            ServiceDefinition previousRoot = context.PreviousRoot;
 
             context.PushProperty("parameters");
 
-            var priorOperationParameters = priorOperation.Parameters.Select(param =>
-                string.IsNullOrWhiteSpace(param.Reference) ? param :
-                FindReferencedParameter(param.Reference, previousRoot.Parameters)
-            );
+            List<(SwaggerParameter param, bool isGlobal)> currParamsResolved =
+                Parameters.Select(param => ResolveParam(param, currentRoot)).ToList();
 
-            var currentOperationParameters = Parameters.Select(param =>
-                string.IsNullOrWhiteSpace(param.Reference) ? param :
-                FindReferencedParameter(param.Reference, currentRoot.Parameters));
+            List<(SwaggerParameter param, bool isGlobal)> priorParamsResolved =
+                priorOperation.Parameters.Select(param => ResolveParam(param, previousRoot)).ToList();
 
-            for (int i = 0; i < currentOperationParameters.Count(); i++)
-            {
-                var curOriginalParameter = Parameters.ElementAt(i);
-                var curParameter = currentOperationParameters.ElementAt(i);
-                curParameter.Extensions.TryGetValue("x-ms-long-running-operation", out var curParameterLocation);
```

Comment:
FYI @mikekistler @mikekistler there was a bug here for 2 years where this:

> curParameter.Extensions.TryGetValue("x-ms-long-running-operation", out var curParameterLocation);

should have been:

> curParameter.Extensions.TryGetValue("x-ms-parameter-location", out var curParameterLocation);

But that was not all, there was also a bug where parameters whose order doesn't matter would cause false positives on parameters whose order matters but didn't actually change. I fixed all of that in this PR